client/src/main/ipc/server.ts

```diff
@@ -131,4 +136,25 @@ export default class IPCServer {
   stop() {
     this.server.close();
   }
+
+  private checkIcon(icon: any): string | undefined {
```

Comment:
FWIW, I think it would have also been okay to just have a single if statement and a generic log statement (like "Icon received does not adhere to requirements") inlined above the call to checkIcon, just to cut down on verbosity.